gtsfm/evaluation/metrics_report.py

```diff
@@ -0,0 +1,198 @@
+            fig.add_trace(
+                go.Bar(x=list(histogram.keys()), y=list(histogram.values()), name=metric_name), row=row, col=col
+            )
+        elif "quartiles" in metric_value[SUMMARY_KEY]:
+            # If all values are available, use them to create box plot.
+            if DATA_KEY in metric_value:
+                fig.add_trace(go.Box(y=metric_value[DATA_KEY], name=metric_name), row=row, col=col)
+            # Else use summary to create box plot.
+            else:
+                quartiles = metric_value[SUMMARY_KEY]["quartiles"]
+                fig.add_trace(
+                    go.Box(
+                        q1=[quartiles["q1"]],
+                        median=[quartiles["q2"]],
+                        q3=[quartiles["q3"]],
+                        lowerfence=[quartiles["q0"]],
+                        upperfence=[quartiles["q4"]],
+                        name=metric_name,
+                    ),
+                    row=row,
+                    col=col,
+                )
+
+    # Return the figure converted to HTML.
+    return fig.to_html(full_html=False, include_plotlyjs="cdn")
+
+
+def get_figures_for_metrics(metrics_group: GtsfmMetricsGroup) -> Tuple[str, str]:
+    """Gets the tables and plots for individual metrics in a metrics group.
+
+    All scalar metrics are reported in the table.
+    Metrics of 1-D distributions have an entry in the table for the mean,
+    and a histogram or box plot as per their property.
+
+    Args:
+        metrics_group: A GtsfmMetricsGroup for any gtsfm module.
+
+    Returns:
+        A tuple of table and plotly figures as HTML code.
+    """
+    scalar_metrics = {}
+    metrics_dict = metrics_group.get_metrics_as_dict()[metrics_group.name]
+
+    # Separate the scalar metrics.
+    for metric, value in metrics_dict.items():
+        if isinstance(value, dict):
+            # Metrics with a dict representation must contain a summary.
+            if not SUMMARY_KEY in value:
+                raise ValueError("Metric {metric} does not contain a summary.")
+            # Add a scalar metric for mean of 1D distributions.
+            scalar_metrics["mean_" + metric] = value[SUMMARY_KEY]["mean"]
+        else:
+            scalar_metrics[metric] = value
+    table = create_table_for_scalar_metrics(scalar_metrics)
+    plots_fig = create_plots_for_distributions(metrics_dict)
+    return table, plots_fig
+
+
+def get_html_metric_heading(metric_name: str) -> str:
+    """Helper to get the HTML heading for a metric name.
+
+    This converts a "metric_name" to "Metric Name" (for readability) and adds some style.
+
+    Returns:
+        HTML for heading as a string.
+    """
+    metric_name = get_readable_metric_name(metric_name)
+    metric_html = f'<p style="font-size:25px;font-family:Arial">{metric_name}</p>'
+    return metric_html
+
+
+def get_html_header() -> str:
+    """Helper to get a HTML header with some CSS styles for tables.""
```

Comment:
can plotly handle this part automatically for us?

Reply:
could you explain the parameter settings a bit/ why they were chosen?

Reply:
> can plotly handle this part automatically for us? 
Unfortunately not, it could if our whole HTML was a plotly figure, but we have tables from tabulate and headings. 

> could you explain the parameter settings a bit/ why they were chosen?
They are mostly arbitrary, or in other words, I found them in some somewhere online and they looked fine. :)

Reply:
Got it. Could you add that high level explanation to the module header docstring? Explaining the file a bit more?

Reply:
Could you please let me know what details you think are interesting at a high level? If youre referring to the HTML header, I dont think that is high level enough to be documented up there.

Reply:
i meant just describing "Plotly cannot render tables, so we create HTML header, and add HTML divisions for tables using the tabulate module, and bar/box charts using Plotly."

Reply:
I would not say "plotly cannot render tables". It does, but it does that as a figure and we cannot select any text. And maybe that will change in future. We dont need to explain dependency details in our documentation. 
I added a sentence about what the HTML contains :)

Reply:
looks good now, marking as resolved